# Relatório Operacional de Pré-Decolagem de uma Nave Fictícia

**Aluno:** Eduardo Lopes da Silveira Mota

**RM:** 563418

**Curso:** Ciência da Computação  
**Disciplina:** Atividade Integradora - Fase 1  
**Data:** 18/03/2026

Este notebook apresenta uma simulação didática de um relatório operacional de pré-decolagem de uma nave fictícia. O objetivo é organizar dados de telemetria, aplicar um algoritmo de verificação, implementar a lógica em Python e interpretar os resultados de forma acadêmica e clara.

## Sumário

- [1. Introdução](#1-introducao)
- [2. Organização e descrição da telemetria](#2-organizacao-e-descricao-da-telemetria)
- [3. Algoritmo de verificação](#3-algoritmo-de-verificacao)
- [4. Implementação em Python](#4-implementacao-em-python)
- [5. Análise energética](#5-analise-energetica)
- [6. Análise assistida por IA](#6-analise-assistida-por-ia)
- [7. Reflexão crítica](#7-reflexao-critica)
- [8. Conclusão](#8-conclusao)

## 1. Introdução

Este trabalho tem como tema a elaboração de um relatório operacional de pré-decolagem de uma nave fictícia. A proposta integra conceitos introdutórios de organização de dados, lógica de programação, tomada de decisão algorítmica e interpretação de resultados computacionais.

A situação simulada considera que sensores de bordo enviam informações importantes antes da partida da nave. Com base nesses dados, um algoritmo deve avaliar se a operação pode prosseguir com segurança.

O notebook foi estruturado de forma acadêmica, com explicações antes das implementações, para facilitar a leitura pelo professor e a exportação em PDF.

## 2. Organização e descrição da telemetria

A telemetria representa o conjunto de informações operacionais coletadas pelos sensores da nave antes da decolagem. Para esta simulação, foram definidos cinco grupos de indicadores conforme especificado na atividade.

### Variáveis monitoradas

| Variável | Descrição | Unidade |
|---|---|---|
| `temperatura_interna_c` | Temperatura interna da cabine | °C |
| `temperatura_externa_c` | Temperatura externa da nave | °C |
| `integridade_estrutural` | Integridade do casco (1 = íntegro, 0 = comprometido) | 0 ou 1 |
| `nivel_energia_percentual` | Nível de carga do sistema de energia principal | % |
| `pressao_tanque_kpa` | Pressão dos tanques de propelente | kPa |
| `modulo_propulsao_ok` | Status do módulo de propulsão | True/False |
| `modulo_navegacao_ok` | Status do módulo de navegação | True/False |
| `modulo_suporte_vida_ok` | Status do módulo de suporte à vida | True/False |

### Faixas seguras de operação

Para fins didáticos, foram definidas as seguintes faixas seguras:

- **Temperatura interna**: entre `18 °C` e `26 °C`
- **Temperatura externa**: entre `-60 °C` e `50 °C`
- **Integridade estrutural**: deve ser `1` (casco íntegro)
- **Nível de energia**: mínimo de `75%`
- **Pressão dos tanques**: entre `300 kPa` e `500 kPa`
- **Módulos críticos**: todos devem estar operacionais (`True`)

Esses limites foram definidos apenas para fins educacionais e não correspondem a parâmetros reais de engenharia aeroespacial.

In [4]:
# Dados de telemetria simulados da nave
telemetria = {
    "temperatura_interna_c": 22,
    "temperatura_externa_c": -45,
    "integridade_estrutural": 1,
    "nivel_energia_percentual": 82,
    "pressao_tanque_kpa": 420,
    "modulo_propulsao_ok": True,
    "modulo_navegacao_ok": True,
    "modulo_suporte_vida_ok": True
}

print("=== TELEMETRIA INICIAL DA NAVE ===")
for nome_variavel, valor_variavel in telemetria.items():
    print(f"  {nome_variavel}: {valor_variavel}")

=== TELEMETRIA INICIAL DA NAVE ===
  temperatura_interna_c: 22
  temperatura_externa_c: -45
  integridade_estrutural: 1
  nivel_energia_percentual: 82
  pressao_tanque_kpa: 420
  modulo_propulsao_ok: True
  modulo_navegacao_ok: True
  modulo_suporte_vida_ok: True


## 3. Algoritmo de verificação

O algoritmo de verificação tem como finalidade decidir entre dois estados finais: `PRONTO PARA DECOLAR` ou `DECOLAGEM ABORTADA`.

A lógica usada neste trabalho é direta: todos os critérios de segurança devem ser atendidos simultaneamente. Se pelo menos um parâmetro crítico estiver fora da faixa definida, a decolagem deve ser abortada.

### Regras de decisão

1. A temperatura interna deve estar entre `18 °C` e `26 °C`.
2. A temperatura externa deve estar entre `-60 °C` e `50 °C`.
3. A integridade estrutural deve ser `1` (casco íntegro).
4. O nível de energia deve ser maior ou igual a `75%`.
5. A pressão dos tanques deve permanecer entre `300 kPa` e `500 kPa`.
6. Todos os módulos críticos (propulsão, navegação e suporte à vida) devem estar operacionais.

### Pseudocódigo

```
INÍCIO
  LEIA telemetria
  motivos_reprovacao ← []

  SE temperatura_interna_c < 18 OU > 26 ENTÃO
    ADICIONE "Temperatura interna fora da faixa segura."
  FIM SE

  SE temperatura_externa_c < -60 OU > 50 ENTÃO
    ADICIONE "Temperatura externa fora da faixa segura."
  FIM SE

  SE integridade_estrutural ≠ 1 ENTÃO
    ADICIONE "Integridade estrutural comprometida."
  FIM SE

  SE nivel_energia_percentual < 75 ENTÃO
    ADICIONE "Nível de energia insuficiente."
  FIM SE

  SE pressao_tanque_kpa < 300 OU > 500 ENTÃO
    ADICIONE "Pressão dos tanques fora da faixa segura."
  FIM SE

  PARA CADA módulo crítico
    SE módulo inoperante ENTÃO
      ADICIONE motivo correspondente
    FIM SE
  FIM PARA

  SE motivos_reprovacao VAZIO ENTÃO
    RETORNE "PRONTO PARA DECOLAR"
  SENÃO
    RETORNE "DECOLAGEM ABORTADA" + motivos
  FIM SE
FIM
```

Se todos os critérios forem satisfeitos, o sistema retorna `PRONTO PARA DECOLAR`. Caso contrário, retorna `DECOLAGEM ABORTADA` e informa os motivos da reprovação.

In [5]:
def verificar_pre_decolagem(dados_telemetria):
    """Analisa a telemetria e retorna o status final da nave e os motivos da decisão."""

    motivos_reprovacao = []

    # Verificação da temperatura interna da cabine
    temp_interna = dados_telemetria["temperatura_interna_c"]
    if temp_interna < 18 or temp_interna > 26:
        motivos_reprovacao.append(
            f"Temperatura interna fora da faixa segura: {temp_interna} °C (esperado: 18–26 °C)."
        )

    # Verificação da temperatura externa
    temp_externa = dados_telemetria["temperatura_externa_c"]
    if temp_externa < -60 or temp_externa > 50:
        motivos_reprovacao.append(
            f"Temperatura externa fora da faixa segura: {temp_externa} °C (esperado: -60 a 50 °C)."
        )

    # Verificação da integridade estrutural
    if dados_telemetria["integridade_estrutural"] != 1:
        motivos_reprovacao.append("Integridade estrutural comprometida (valor 0 detectado).")

    # Verificação do nível de energia
    nivel_energia = dados_telemetria["nivel_energia_percentual"]
    if nivel_energia < 75:
        motivos_reprovacao.append(
            f"Nível de energia insuficiente: {nivel_energia}% (mínimo: 75%)."
        )

    # Verificação da pressão dos tanques
    pressao = dados_telemetria["pressao_tanque_kpa"]
    if pressao < 300 or pressao > 500:
        motivos_reprovacao.append(
            f"Pressão dos tanques fora da faixa segura: {pressao} kPa (esperado: 300–500 kPa)."
        )

    # Verificação dos módulos críticos
    modulos = {
        "modulo_propulsao_ok": "Módulo de propulsão",
        "modulo_navegacao_ok": "Módulo de navegação",
        "modulo_suporte_vida_ok": "Módulo de suporte à vida"
    }
    for chave, nome in modulos.items():
        if not dados_telemetria[chave]:
            motivos_reprovacao.append(f"{nome} inoperante.")

    if len(motivos_reprovacao) == 0:
        status_final = "PRONTO PARA DECOLAR"
    else:
        status_final = "DECOLAGEM ABORTADA"

    return status_final, motivos_reprovacao


status_final, motivos_reprovacao = verificar_pre_decolagem(telemetria)

print("=== RESULTADO DA VERIFICAÇÃO OPERACIONAL ===")
print(f"Status final: {status_final}")

if len(motivos_reprovacao) == 0:
    print("Todos os parâmetros estão dentro dos limites definidos.")
else:
    print("Motivos da abortagem:")
    for indice, motivo in enumerate(motivos_reprovacao, start=1):
        print(f"  {indice}. {motivo}")

=== RESULTADO DA VERIFICAÇÃO OPERACIONAL ===
Status final: PRONTO PARA DECOLAR
Todos os parâmetros estão dentro dos limites definidos.


## 4. Implementação em Python

A implementação foi realizada em Python e foi dividida em três etapas: definição dos dados de telemetria, função de verificação com regras de segurança e exibição dos resultados.

Essa organização facilita a compreensão do fluxo lógico do programa e permite a alteração dos valores para testar diferentes cenários de pré-decolagem. A saída do script indica claramente se a nave está pronta ou se a decolagem deve ser abortada, acompanhada dos motivos em caso de reprovação.

## 5. Análise energética

Além da verificação geral de segurança, foi realizada uma análise energética para calcular a autonomia inicial da nave e verificar se a energia disponível é suficiente para suportar a decolagem.

### Parâmetros considerados

| Parâmetro | Valor adotado |
|---|---|
| Capacidade total do sistema de energia | `500 kWh` |
| Carga atual | `82%` (conforme telemetria) |
| Consumo estimado na decolagem | `120 kWh` |
| Taxa de perdas energéticas | `8%` da energia disponível |

### Fórmulas utilizadas

- **Energia disponível** = Capacidade total × (Carga atual / 100)
- **Perdas energéticas** = Energia disponível × Taxa de perda
- **Energia líquida** = Energia disponível − Perdas energéticas
- **Autonomia após decolagem** = Energia líquida − Consumo estimado na decolagem

In [6]:
# Parâmetros da análise energética
capacidade_total_kwh = 500                                        # Capacidade total do sistema (kWh)
carga_atual_percentual = telemetria["nivel_energia_percentual"]   # 82%
consumo_decolagem_kwh = 120                                       # Consumo estimado na decolagem (kWh)
taxa_perda_energetica = 0.08                                      # 8% de perdas energéticas

# Cálculos
energia_disponivel_kwh = capacidade_total_kwh * (carga_atual_percentual / 100)
perdas_energeticas_kwh = energia_disponivel_kwh * taxa_perda_energetica
energia_liquida_kwh = energia_disponivel_kwh - perdas_energeticas_kwh
autonomia_pos_decolagem_kwh = energia_liquida_kwh - consumo_decolagem_kwh

print("=== ANÁLISE ENERGÉTICA ===")
print(f"  Capacidade total do sistema:      {capacidade_total_kwh:.1f} kWh")
print(f"  Carga atual ({carga_atual_percentual}%):               {energia_disponivel_kwh:.1f} kWh")
print(f"  Perdas energéticas (8%):          -{perdas_energeticas_kwh:.1f} kWh")
print(f"  Energia líquida disponível:       {energia_liquida_kwh:.1f} kWh")
print(f"  Consumo estimado na decolagem:    -{consumo_decolagem_kwh:.1f} kWh")
print(f"  Autonomia após a decolagem:       {autonomia_pos_decolagem_kwh:.1f} kWh")
print()

if autonomia_pos_decolagem_kwh > 0:
    print("Resultado energético: ENERGIA SUFICIENTE PARA DECOLAGEM E OPERAÇÃO INICIAL.")
else:
    print("Resultado energético: ENERGIA INSUFICIENTE — decolagem energeticamente inviável.")

=== ANÁLISE ENERGÉTICA ===
  Capacidade total do sistema:      500.0 kWh
  Carga atual (82%):               410.0 kWh
  Perdas energéticas (8%):          -32.8 kWh
  Energia líquida disponível:       377.2 kWh
  Consumo estimado na decolagem:    -120.0 kWh
  Autonomia após a decolagem:       257.2 kWh

Resultado energético: ENERGIA SUFICIENTE PARA DECOLAGEM E OPERAÇÃO INICIAL.


### Interpretação da análise energética

Com a carga atual de **82%** sobre uma capacidade total de **500 kWh**, a nave dispõe de **410 kWh** brutos. Após descontar as perdas energéticas estimadas em **8% (32,8 kWh)**, a energia líquida disponível é de **377,2 kWh**. Subtraindo o consumo estimado de **120 kWh** para a decolagem, a autonomia remanescente é de **257,2 kWh** — margem considerada suficiente para as operações iniciais da missão.

Em um cenário real, essa análise seria muito mais complexa, envolvendo diferentes subsistemas, redundâncias e fatores ambientais variáveis.

## 6. Análise assistida por IA

A análise assistida por inteligência artificial foi realizada solicitando à IA a classificação dos dados de telemetria, a identificação de possíveis anomalias e sugestões de risco com base nos valores coletados pelos sensores da nave.

### Prompt enviado à IA

> *"Analise os seguintes dados de telemetria de uma nave fictícia antes da decolagem e forneça: (1) classificação de cada parâmetro como NORMAL, ALERTA ou CRÍTICO; (2) identificação de possíveis anomalias; (3) sugestões de risco. Dados: temperatura interna 22 °C, temperatura externa -45 °C, integridade estrutural 1, nível de energia 82%, pressão dos tanques 420 kPa, todos os módulos operacionais."*

### Classificação dos parâmetros

| Parâmetro | Valor | Faixa segura | Classificação |
|---|---|---|---|
| Temperatura interna | 22 °C | 18–26 °C | ✅ NORMAL |
| Temperatura externa | -45 °C | -60 a 50 °C | ⚠️ ALERTA |
| Integridade estrutural | 1 | 1 | ✅ NORMAL |
| Nível de energia | 82% | ≥ 75% | ✅ NORMAL |
| Pressão dos tanques | 420 kPa | 300–500 kPa | ✅ NORMAL |
| Módulo de propulsão | Operacional | Operacional | ✅ NORMAL |
| Módulo de navegação | Operacional | Operacional | ✅ NORMAL |
| Módulo de suporte à vida | Operacional | Operacional | ✅ NORMAL |

### Identificação de possíveis anomalias

Com base nos dados simulados, a IA identificou que a maioria dos parâmetros está dentro das faixas seguras. No entanto, foram apontadas as seguintes observações preventivas:

- **Temperatura externa de -45 °C** (⚠️ ALERTA): embora dentro da faixa segura, o valor está relativamente próximo ao limite inferior de -60 °C. Variações bruscas durante a ascensão podem aproximar a leitura do limite crítico. Recomenda-se monitoramento contínuo.
- **Pressão dos tanques em 420 kPa**: valor na metade superior da faixa (300–500 kPa). Nenhuma anomalia detectada, porém merece atenção em caso de variação durante a aceleração inicial.
- **Nível de energia em 82%**: dentro dos parâmetros, mas a margem de apenas 7% acima do mínimo exigido (75%) pode ser insuficiente em cenários com consumo inesperado ou falha de subsistema.

### Sugestões de risco

1. **Risco térmico externo**: a temperatura de -45 °C pode gerar tensões mecânicas em componentes externos e vedações do casco. Recomenda-se verificar o isolamento térmico antes da autorização final de decolagem.
2. **Risco energético**: a margem de 7% sobre o mínimo de energia requerido pode ser insuficiente em emergências. Considerar recarga complementar antes da decolagem.
3. **Risco de pressão nos tanques**: variações de pressão durante a aceleração inicial merecem monitoramento em tempo real para evitar falhas estruturais nos tanques de propelente.

### Observação acadêmica

É importante ressaltar que toda análise produzida por IA deve ser verificada e validada pelo responsável técnico. A IA serve como ferramenta de apoio à decisão, não como substituta do julgamento humano qualificado. Todo conteúdo gerado foi revisado e validado por mim antes da inclusão neste relatório.

## 7. Reflexão crítica

### Ética e responsabilidade

A simulação de um sistema de pré-decolagem, mesmo em contexto acadêmico, convida a uma reflexão sobre as responsabilidades éticas envolvidas no desenvolvimento de tecnologias críticas. Em sistemas reais, um erro de algoritmo ou uma falha de sensor pode resultar em perdas humanas e materiais irreparáveis.

Engenheiros e cientistas da computação que atuam nessa área carregam uma responsabilidade que vai além do código: precisam garantir que os sistemas sejam seguros, auditáveis e transparentes. A adoção de critérios claros de tomada de decisão, como os utilizados neste notebook, é um princípio ético fundamental, pois permite que qualquer pessoa entenda por que uma decolagem foi aprovada ou abortada, sem depender de uma "caixa preta" inacessível. A transparência algorítmica não é apenas uma boa prática de engenharia; é uma exigência ética quando vidas humanas estão em jogo.

### Impacto social da exploração espacial

A exploração espacial gera impactos sociais profundos e contraditórios. Por um lado, impulsiona avanços tecnológicos que beneficiam a sociedade: satélites de comunicação, previsão do tempo, monitoramento ambiental e desenvolvimento de novos materiais são alguns exemplos diretos. Por outro, o alto custo das missões espaciais levanta questões sobre prioridades: enquanto bilhões são investidos em foguetes, grande parte da população mundial ainda carece de acesso a saneamento básico, educação de qualidade e saúde.

Nesse sentido, a exploração espacial só se justifica socialmente quando seus benefícios são democratizados e quando há esforço genuíno em tornar a ciência acessível para além das nações mais ricas. O desenvolvimento de tecnologias espaciais precisa caminhar junto com a responsabilidade de reduzir desigualdades, e não aprofundá-las. Programas que incluem países em desenvolvimento, educação científica desde a infância e compartilhamento aberto de dados são passos nessa direção.

### Sustentabilidade tecnológica

A sustentabilidade tecnológica é um dos maiores desafios da corrida espacial contemporânea. O descarte de estágios de foguetes, a geração de detritos orbitais e o consumo de combustíveis fósseis em lançamentos são problemas reais que ameaçam tanto a operabilidade do espaço próximo quanto o ambiente terrestre.

Iniciativas como o desenvolvimento de foguetes reutilizáveis e o uso de propelentes menos poluentes representam avanços importantes. No entanto, a sustentabilidade tecnológica exige uma visão sistêmica: não basta desenvolver tecnologias eficientes se os modelos de negócio e as políticas públicas não incentivarem práticas responsáveis de uso e descarte. A construção deste notebook reforça a ideia de que a educação tecnológica precisa incorporar essas dimensões críticas desde o início da formação, preparando profissionais capazes de tomar decisões técnicas com consciência de seus impactos mais amplos.

## 8. Conclusão

Este notebook apresentou uma simulação de relatório operacional de pré-decolagem de uma nave fictícia, integrando telemetria, algoritmo de verificação, implementação em Python, análise energética, análise assistida por IA e reflexão crítica.

Os dados monitorados: temperatura interna e externa, integridade estrutural, níveis de energia, pressão dos tanques e status dos módulos críticos foram organizados em um dicionário Python e avaliados por uma função de verificação com regras de segurança bem definidas. A análise energética demonstrou como calcular a autonomia inicial considerando capacidade total em kWh, carga atual, consumo estimado na decolagem e perdas energéticas.

A atividade demonstrou como regras lógicas simples podem apoiar uma decisão automatizada, ao mesmo tempo em que evidenciou a importância da clareza metodológica, da responsabilidade ética e da reflexão crítica sobre os impactos sociais e ambientais da tecnologia. Dessa forma, o trabalho atende ao objetivo de articular conteúdos introdutórios de programação em um formato acadêmico, organizado e pronto para documentação em notebook.